### PR TITLE
feat(dpp): identity facade

### PR DESCRIPTION
**Pull request by markin-io**
## Issue being fixed or feature implemented

Implements IdentityFacade for DashPlatformProtocol

## What was done?

- IdentityFactory bindings and tests
- IdentityValidator bindings and tests
- IdentityFacade bindings and tests

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/wasm-dpp/src/identity/identity_facade.rs [hunk 1]
```diff
@@ -1,46 +1,278 @@
...
+    ) -> Result<IdentityWasm, JsValue> {
+        let options: FromObjectOptions = if options.is_object() {
+            with_js_error!(serde_wasm_bindgen::from_value(options))?
+        } else {
+            Default::default()
+        };
+
+        let identity_json = utils::stringify(&identity_object)?;
+        let raw_identity: Value =
+            serde_json::from_str(&identity_json).map_err(|e| e.to_string())?;
```
Comment (qrayven) [inline packages/wasm-dpp/src/identity/identity_facade.rs:90]: these two lines could be replaces with `raw_identity.with_serde_to_json_value()`  from `ToSerdeJsonExt`

File: packages/wasm-dpp/src/identity/identity_factory.rs [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+        if js_sys::Reflect::has(public_keys, &"add".into()).unwrap_or(false) {
+            let raw_add_public_keys = js_sys::Reflect::get(public_keys, &"add".into()).unwrap();
+
+            let add_public_keys_array: &js_sys::Array = raw_add_public_keys
+                .dyn_ref::<js_sys::Array>()
+                .ok_or_else(|| {
+                    RustConversionError::Error(String::from("public keys to add must be array"))
+                        .to_js_value()
+                })?;
```
Comment (qrayven) [inline packages/wasm-dpp/src/identity/identity_factory.rs:211]: NIT:  extraction array from `add` and `disable`. seems duplicates. Would be possible to extract the functionality to one function?

**Review by qrayven — DISMISSED** [review]

File: packages/wasm-dpp/src/identity/identity_facade.rs (hunk 1 above)
Comment (markin-io) [reply packages/wasm-dpp/src/identity/identity_facade.rs]: Fixed

File: packages/wasm-dpp/src/identity/identity_factory.rs (hunk 2 above)
Comment (markin-io) [reply packages/wasm-dpp/src/identity/identity_factory.rs]: Fixed

**Review by antouhou — APPROVED** [review]
👍

**Pull request merged**